tools/project.py

```diff
@@ -46,7 +46,7 @@ def setup_project(ide, target, program=None, source_dir=None, build=None, export
             project_name = TESTS[program]
         else:
             project_name = basename(normpath(realpath(source_dir[0])))
-        src_paths = source_dir
+        src_paths = {relpath(path, project_dir): [path] for path in source_dir}
```

Comment:
What happens if project_dir gets set to export_path but export_path (build argument) doesn't exist?

Reply:
Relpath will still work correctly. According to [the docs](https://docs.python.org/3/library/os.path.html?highlight=relpath#os.path.relpath); no fs access is done to verify the existence of the path.

Reply:
But what about later when the exporter tries to put files there?

Reply:
I'll just check this out and try it. Give me a bit.